### ISSUE TITLE
Is it possible to dynamically set the rolling session duration in the callback?

### ISSUE THREAD

**Issue by tachang** [auth0/nextjs-auth0#1034]
### Checklist

- [X] I have looked into the [README](https://github.com/auth0/nextjs-auth0#readme) and have not found a suitable solution or answer.
- [X] I have looked into the [examples](https://github.com/auth0/nextjs-auth0/blob/main/EXAMPLES.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/nextjs-auth0/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/nextjs-auth0/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com/c/sdks/5) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Describe the problem you'd like to have solved

The ability to specify individual session idle timeouts depending on what is in the claim.

### Describe the ideal solution

Being able to access the auth0 configuration object in the callback handler.

**Comment by adamjmcgrath** [comment]
Hi @tachang - thanks for raising this

This would not be possible since you wouldn't be able to decode the token until you instantiated the SDK. Also you couldn't just do it in the callback handler, you'd need to do it for every request.

You could set a large rolling session duration in the SDKs configuration, then use your own properties on the session to decide if the session is expired - but this would be out of scope of the SDK

**Comment by tachang** [comment]
@adamjmcgrath Would you be open to a PR to enable access to the cookie store in which you can access the configuration?

This is an example of what I was thinking: https://github.com/tachang/nextjs-auth0/commit/c39e8cd7f5fb1de960eac33ba1fab2a886615f11

**Comment by adamjmcgrath** [comment]
> @adamjmcgrath Would you be open to a PR to enable access to the cookie store in which you can access the configuration?
> This is an example of what I was thinking: https://github.com/tachang/nextjs-auth0/commit/c39e8cd7f5fb1de960eac33ba1fab2a886615f11

Thanks for the offer @tachang, but we don't want to expose the underlying cookie store in the public api. We go to great lengths to make sure this SDK is only used as an openid connect client and this would open up a bunch of new use cases we don't want to support.

> So you'd have the callback handler set the rollingDuration based on claims and then when a user was authenticated you would inspect the configuration again and set it as needed.

No, as I said previously, you wouldn't be able to decode the token until you instantiated the SDK. So configuring the SDK based on the claims of the token would be impossible - unless you did all the work of acquiring/decoding the token yourself before instantiating the SDK (and decrypting the session before instantiating the SDK on every authenticated request).

In short, this use case is not possible with this SDK I'm afraid. I would either go back to the problem you're originally trying to solve and come up with a different solution, fork this SDK, or use something else.

**Comment by tachang** [comment]
> No, as I said previously, you wouldn't be able to decode the token until you instantiated the SDK. So configuring the SDK based on the claims of the token would be impossible - unless you did all the work of acquiring/decoding the token yourself before instantiating the SDK (and decrypting the session before instantiating the SDK on every authenticated request).

It seemed to have work after implementing the code aginst the v.1.9.2 branch. I would edit the rollingsessiontimeout in the aftercallback handler which is done before the `exp` is calculated. This results in the jwt having the correct dynamic expiration.

It just required access to the underlying configuration off what was returned by InitAuth0.

**Comment by tachang** [comment]
I do understand forking is always possible but still would like to explore the possibility of a change like this going into the mainline.

**Comment by tachang** [comment]
How about a method to update the configuration after initialization?

**Comment by adamjmcgrath** [comment]
> How about a method to update the configuration after initialization?

Thanks for the suggestion @tachang. Changing the config after initialization would require reinitializing the whole SDK, since a bunch of state is set up in closures at initialization time, and we already have a method for that [`initAuth0`](https://auth0.github.io/nextjs-auth0/types/index.InitAuth0.html)

**Comment by adamjmcgrath** [comment]
Thanks for sharing your workaround @tachang - this doesn't look like a use case we can support, but I'll continue to monitor demand for any similar use cases and also any other suggestions that might help your use case. Closing as I don't think there's anything else for the SDK to do.
